### PR TITLE
Draft: Database tables for Milestone and Reward

### PR DESCRIPTION
**Pull request by benbacardi**

### CODE REVIEW

**Review by dwrss — APPROVED** [review]

File: Shared/Networking/Model.swift [hunk 1]
```diff
@@ -62,25 +90,27 @@ struct TiltifyCampaignRewardImage: Codable {
...
     let slug: String
     let status: String
     let team: String?
+    let user: TiltifyUser
     let description: String
     let totalAmountRaised: TiltifyAmount
     let name: String
     let originalGoal: TiltifyAmount
-    let rewards: [TiltifyCampaignReward]
+    let rewards: [TiltifyCampaignReward]
```
Comment (dwrss) [inline Shared/Networking/Model.swift:113]: Is the whitespace at the end intentional?

Comment (benbacardi) [reply Shared/Networking/Model.swift:113]: Nope!

**Review by dwrss — APPROVED** [review]

File: Shared/Model/DB/AppDatabase.swift [hunk 2]
```diff
@@ -153,6 +153,10 @@ extension AppDatabase {
         }
     }
     
+    private func fetchCampaign(with id: UUID, using db: Database) throws -> Campaign? {
```
Comment (dwrss) [inline Shared/Model/DB/AppDatabase.swift:156]: Might want to put the `db` parameter first (or swap the other methods) for consistency.

File: Shared/Views/CampaignView.swift [hunk 3]
```diff
@@ -159,34 +201,198 @@ struct CampaignView: View {
...
+                fetchTask?.cancel()
+                fetchTask = Task {
+                    await fetch()
+                }
+            }
+            
+            // New Database and API Fetch!
+            Task {
+                await fetch()
+            }
```
[the remark this answers is not in the document]
Comment (benbacardi) [reply Shared/Views/CampaignView.swift:228]: That's good to know! 👍

File: Shared/Model/DB/AppDatabase.swift (hunk 2 above)
Comment (benbacardi) [reply Shared/Model/DB/AppDatabase.swift:156]: Makes sense 👍

**Pull request merged**